### PR TITLE
[BUGFIX canary]: Remove the last Symbols

### PR DESCRIPTION
**Pull request by dcyriller**
### Description

This is an attempt at removing the last `Symbol`s from master. The goal is to fix IE compatibility regarding Symbol support, as detailed in this issue: https://github.com/emberjs/data/issues/6380. 

It won't have to be backported. Indeed, the Identifier feature has been turned on in https://github.com/emberjs/data/pull/6366 and included in a tag for the first time in `v3.14.0-alpha.2`.

Thank you for pointing me to that solution @rwjblue (https://github.com/emberjs/data/pull/6389).

### Concerns

With the implementation I've done, I'm concerned by the fact:
- it kind of changes the `StableIdentifier` interface
- the `isStableIdentifier` signature is modified

But as it is still on canary, it might be fine?

Is the `symbol` fix good enough? It has been suggested it could be replaced by the WeakMap fix every time `symbol` is used.

### Tests

I've run the tests with the command: `yarn test`. Identifier feature flag being turned on, the test seemed to be fine. I'd be happy to add more if needed.

### CODE REVIEW

**Comment by dcyriller** [comment]
External Partner Travis is failing in Azure, but not in Travis.

File: packages/store/addon/-private/ts-interfaces/identifier.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 /**
   @module @ember-data/store
 */
-export const IS_IDENTIFIER = Symbol('is-identifier');
+import { symbol } from '../ts-interfaces/utils/symbol';
+
+export const IDENTIFIERS = new WeakMap();
```
Comment (rwjblue) [inline packages/store/addon/-private/ts-interfaces/identifier.ts]: Nothing in here uses the WeakMap, can you move the WeakMap itself into the location of the utility methods that use it and not export it? (either moving the weakmap there, or moving those helper methods here)

Comment (dcyriller) [reply packages/store/addon/-private/ts-interfaces/identifier.ts]: Makes sense! Thanks, it's done.

**Comment by rwjblue** [comment]
Ya, if there are for sure places where we know we can remove, we totally should.

File: packages/-ember-data/tests/integration/identifiers/cache-test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      } catch (e) {
+        assert.equal(
+          e.message,
+          'The lid generated for the new record is not unique as it matches an existing identifier'
+        );
+      }
+
+      cache.forgetRecordIdentifier(identifier);
+
+      const regeneratedIdentifier = cache.createIdentifierForNewRecord(houseHash);
```
[the remark this answers is not in the document]
Comment (dcyriller) [reply packages/-ember-data/tests/integration/identifiers/cache-test.ts]: done, I ended up splitting the test into two tests

**Pull request merged**